src/main/java/htsjdk/samtools/SamReaderFactory.java

```diff
@@ -311,7 +311,24 @@ public SamReader open(final SamInputResource resource) {
                     if (SamStreams.isBAMFile(bufferedStream)) {
```

Comment:
Is it ok to leave the `indexMaybe.asFile()` call intact here (even if the index might be in a bucket)?